drools-ansible-rulebook-integration-api/src/test/java/org/drools/ansible/rulebook/integration/api/toexecmodel/MapConditionToPatternTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.ansible.rulebook.integration.api.toexecmodel;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.drools.ansible.rulebook.integration.api.domain.RuleGenerationContext;
+import org.drools.ansible.rulebook.integration.api.domain.conditions.MapCondition;
+import org.drools.ansible.rulebook.integration.api.rulesmodel.RulesModelUtil;
+import org.drools.core.facttemplates.Event;
+import org.drools.model.PatternDSL;
+import org.drools.model.functions.Predicate1;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MapConditionToPatternTest {
+
+    @Test
+    public void selectExpression() throws Exception {
+        // Create MapCondition
+        LinkedHashMap<Object, Object> lhsValueMap = createLhsForEventField("levels");
+        LinkedHashMap<Object, Object> rhsValueMap = createRhsWithOperatorAndValue("String", ">", "Integer", 25);
+        LinkedHashMap<Object, Object> selectExpression = createSelectExpression(lhsValueMap, rhsValueMap);
+        LinkedHashMap<Object, Object> rootMap = createAllCondition(selectExpression);
+        MapCondition mapCondition = new MapCondition(rootMap);
```

Comment:
I may be able to create better util methods to compose a `MapCondition` while adding more tests.